src/analysis/retail/priest/shadow/modules/talents/ShadowCrash.tsx

```diff
@@ -40,7 +55,28 @@ class ShadowCrash extends Analyzer {
     );
   }
 
+  onCast() {
+    this.cast += 1;
+  }
+
   onDamage(event: DamageEvent) {
+    // If the damage occurs before the first cast
+    // Then we fabricate a cast of shadowcrash.
+    // Shadow Crash always takes 1.5 seconds to deal damage after the cast
+    if (this.cast === 0) {
+      this.eventEmitter.fabricateEvent(
```

Comment:
This can be done by adding `damageEventIds` to the declaration in the Abilities.tsx like so:
```
      {
        spell: TALENTS_DEMON_HUNTER.THE_HUNT_TALENT.id,
        enabled: combatant.hasTalent(TALENTS_DEMON_HUNTER.THE_HUNT_TALENT.id),
        category: SPELL_CATEGORY.COOLDOWNS,
        cooldown: 90,
        gcd: {
          base: 1500,
        },
        castEfficiency: {
          suggestion: true,
          recommendedEfficiency: 0.8,
          extraSuggestion: (
            <>
              The only time you should delay casting{' '}
              <SpellLink id={TALENTS_DEMON_HUNTER.THE_HUNT_TALENT.id} /> is when you're expecting
              adds to spawn soon.
            </>
          ),
        },
        damageSpellIds: [SPELLS.THE_HUNT_CHARGE.id, SPELLS.THE_HUNT_DOT.id],
      },
```

The framework should automatically handle creating the pre-cast event using this method.

Reply:
That is much easier.  I've added the damageSpellIds in abilities and removed the fabricateEvent from ShadowCrash.